### PR TITLE
update staging and deploy scripts

### LINKED ISSUE #9: Setup deploy task
Need to setup a staging subdomain and production subdomain to deploy to.

### PR DESCRIPTION
**Pull request by mcbouslog**
Closes #9.

Related static files:
- [x] https://github.com/zooniverse/static/blob/master/sites/star.lab-preview.zooniverse.org.conf
- [ ] sites/lab.zooniverse.org.conf

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,11 +19,16 @@
     "zooniverse"
   ],
   "scripts": {
-    "start": "NODE_ENV=development check-engines && check-dependencies && webpack-dashboard -p 3777 -- babel-node server.js",
-    "test": "NODE_ENV=development BABEL_ENV=test mocha",
+    "start": "export NODE_ENV=development; check-engines && check-dependencies && webpack-dashboard -- babel-node ./server.js",
+    "_build": "HEAD_COMMIT=$(git rev-parse --short HEAD); check-engines && check-dependencies && rimraf dist; webpack --config ./webpack.production.config.js -p",
```
Comment (mcbouslog) [inline package.json]: `rimraf` needed if using CleanWebpackPlugin in [webpack.production.config](https://github.com/zooniverse/pfe-lab/blob/master/webpack.production.config.js#L21)? I think duplicative, can omit rimraf here in script.

File: package.json [hunk 2]
```diff
@@ -19,11 +19,16 @@
...
     "eslint": "eslint .",
-    "build": "webpack --config webpack.production.config.js -p",
-    "stage": "NODE_ENV=staging npm run build && publisssh ./dist zooniverse-static/preview.zooniverse.org/pfe-lab"
+    "_publish": "publisssh ./dist \"zooniverse-static/$PATH_ROOT/$SUBDIR\"",
+    "_log-staging-url": "echo \"Staged at https://$SUBDIR.lab-preview.zooniverse.org/\"",
+    "_stage": "export PATH_ROOT=lab-preview.zooniverse.org/pfe-lab; npm run _build && npm run _publish && npm run _log-staging-url",
```
Comment (mcbouslog) [inline package.json]: per the static file I think the `PATH-ROOT` should just be `preview.zooniverse.org/pfe-lab`

**Comment by mcbouslog** [comment]
doesn't appear I'm handling `SUBDIR`correctly, though PFE also appears off:
 
- pfe-lab (expected `Staged at https://update-deploy-scripts.lab-preview.zooniverse.org/`)
screen shot 2017-07-11 at 11 26 44 am

- PFE
screen shot 2017-07-11 at 11 26 56 am

**Comment by mcbouslog** [comment]
I'm unsure if/how the PFE Lab staging doorkeeper callback urls should be updated to reflect the dynamic branch reference in the beginning of the url.

**Comment by mcbouslog** [comment]
Build and deploy appear successful via console:

screen shot 2017-07-12 at 11 35 10 am

Get following error on page load:

screen shot 2017-07-12 at 11 32 52 am

**Comment by srallen** [comment]
@mcbouslog This has to do with the SSL certificate configuration, not the deployment script. Not sure how to fix that, since I haven't worked on that before for our domains.

**Comment by astopy** [comment]
Oops, that should be fixed now.

**Pull request merged**